x-pack/plugins/observability_solution/observability_ai_assistant/common/utils/process_openai_stream.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import OpenAI from 'openai';
+import { filter, map, Observable, tap } from 'rxjs';
+import { v4 } from 'uuid';
+import type { Logger } from '@kbn/logging';
+import { Message } from '..';
+import {
+  type ChatCompletionChunkEvent,
+  createInternalServerError,
+  createTokenLimitReachedError,
+  StreamingChatResponseEventType,
+} from '../conversation_complete';
+
+export type CreateChatCompletionResponseChunk = OpenAI.ChatCompletionChunk;
+
+export function processOpenAiStream(logger: Logger) {
```

Comment:
What's the difference between this, and the adapter called `processOpenAiStream`?

https://github.com/elastic/kibana/blob/b20b261c4f399532e16fb247fb0a719093dd7e6a/x-pack/plugins/observability_solution/observability_ai_assistant/server/service/client/adapters/process_openai_stream.ts#L30

Reply:
oof, good call, I think this is a merge conflict gone awry. Thanks for spotting it, I'll follow up.

Reply:
ok, so at least it's functionally still the same - the relevant changes are in both files, and there's already a test for it. still, will delete it in a follow-up PR.